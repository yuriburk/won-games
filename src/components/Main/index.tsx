type MainProps = {
  title?: string
}

const Main = ({ title = 'React AvanÃ§ado' }: MainProps) => (
  <main className="flex items-center justify-center bg-background text-white w-screen h-screen p-12 text-center">
    <h1>{title}</h1>
  </main>
)

export default Main
