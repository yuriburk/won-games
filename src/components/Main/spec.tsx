import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should be able to render the heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react avanÃ§ado/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should be able to render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveClass('bg-background')
  })
})
