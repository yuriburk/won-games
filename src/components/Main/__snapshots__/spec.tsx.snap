// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should be able to render the heading 1`] = `
<main
  class="flex items-center justify-center bg-background text-white w-screen h-screen p-12 text-center"
>
  <h1>
    React Avan√ßado
  </h1>
</main>
`;
